script/zmb

```diff
@@ -0,0 +1,424 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use feature 'say';
+
+use Getopt::Long qw( GetOptionsFromArray :config require_order );
+
+use Pod::Usage;
+
+use JSON::PP;
+use LWP::UserAgent;
+
+=head1 NAME
+
+B<zmb> - Shell bindings for the Zonemaster::Backend RPC API
+
+Zmb is meant to be pronounced I<Zimba>.
+
+=head1 SYNOPSIS
+
+zmb COMMAND [OPTIONS]
+
+=head1 GLOBAL OPTIONS
+
+ --help        Show usage
+ --verbose     Show RPC query
+
+=cut
+
+sub main {
+    my @argv = @_;
+
+    my $opt_help;
+    my $opt_verbose;
+    GetOptionsFromArray(
+        \@argv,
+        'help'    => \$opt_help,
+        'verbose' => \$opt_verbose,
+    ) or pod2usage( 2 );
+    if ( !@argv ) {
+        pod2usage( -verbose => 99, -sections => ['SYNOPSIS', 'GLOBAL OPTIONS'], -exitval => 'NOEXIT' );
+        show_commands();
+        exit 1;
+    }
+    my $cmd = shift @argv;
+    pod2usage( 1 ) if !defined $cmd;
+    my $cmd_sub = \&{ "cmd_" . $cmd };
+    pod2usage( "'$cmd' is not a command" ) if !defined &$cmd_sub;
+    pod2usage( -verbose => 99, -sections => ["COMMANDS/$cmd"] ) if $opt_help;
+
+    my $json = &$cmd_sub( @argv );
+
+    if ( $json ) {
+        say $json if $opt_verbose;
+        my $request  = to_request( $json );
+        my $response = submit( $request );
+        say $response;
+    }
+}
+
+
+=head1 COMMANDS
+
+=head2 man
+
+Show the full manual page.
+
+    zmb man
+
+=cut
+
+sub cmd_man {
+    pod2usage( -verbose => 2 );
+}
+
+
+=head2 non_existing_method
+
+Call a non-existing RPC method.
+
+    zmb non_existing_method
+
+=cut
+
+sub cmd_non_existing_method {
+    return to_jsonrpc(
+        id     => 1,
+        method => 'non_existing_method',
+    );
+}
+
+
+=head2 version_info
+
+    zmb version_info
+
+=cut
+
+sub cmd_version_info {
+    return to_jsonrpc(
+        id     => 1,
+        method => 'version_info',
+    );
+}
+
+
+=head2 start_domain_test
+
+Call start_domain_test.
+
+    zmb start_domain_test [OPTIONS]
+
+Options:
+
+    --domain DOMAIN_NAME
+    --nameserver DOMAIN_NAME:IP_ADDRESS
+    --client-id CLIENT_ID
+    --client-version CLIENT_VERSION
+    --ds-info DS_INFO
```

Comment:
State what options that are mandatory.

Reply:
All of them are optional by design. If zmb crashes because you omitted some of them, that's a bug in zmb.

Reply:
You cannot run a test without domain name.

IPv4/6 on/off is missing.